.github/workflows/common.yml

```diff
@@ -104,6 +104,19 @@ jobs:
         uses: codecov/test-results-action@v1
         with:
           token: ${{ secrets.CODECOV_TOKEN }}
+      - name: Slack Notification on Failure
```

Comment:
The Slack notification configuration is duplicated between common.yml and examples.yml. Consider extracting this into a reusable workflow to maintain DRY principles and make future updates easier. Example:

```yaml
# .github/workflows/slack-notify.yml
name: Slack Notification
on:
  workflow_call:
    inputs:
      title:
        required: true
        type: string
      mentions:
        required: true
        type: string
      message:
        required: true
        type: string
    secrets:
      SLACK_WEBHOOK:
        required: true

jobs:
  notify:
    runs-on: ubuntu-latest
    steps:
      - uses: rtCamp/action-slack-notify@v2
        env:
          SLACK_WEBHOOK: ${{ secrets.SLACK_WEBHOOK }}
          SLACK_TITLE: ${{ inputs.title }}
          SLACK_MESSAGE: ${{ inputs.mentions }} ${{ inputs.message }}
          # ... other common configurations
```